### DISCUSSION TITLE
Mac- Running webdriverio automation on electron app

### DISCUSSION THREAD

**Discussion by ellatom** [webdriverio/webdriverio#8199] [Technical Discussion]
Hi,
I want to run automation on electron app on mac.
I have set "binaryPath" to the location of the app on mac, however the app won't open and all tests are failing(I have checked. and manually clicking on the app it's being opened).
When I run the tests on Windows OS the tests are successful and don't fail.
what could be the reason?
This is my config file:
```
{
  capabilities: [
    {
      acceptInsecureCerts: true,
    },
  ],

  services: [
    [
      'electron',
      {
        newSessionPerTest: false,
        binaryPath: process.env.RELATIVE_PATH_TO_ELECTRON,

        appArgs: ['--disable-infobars', '--disable-dev-shm-usage', '--no-sandbox', '--remote-debugging-port=9222'],
        chromedriver: {
          port: 9519,
          logFileName: 'wdio-chromedriver.log',
        },
        safaridriver: {
          port: 4444,
          logFileName: 'wdio-safaridriver.log',
        },
      },
    ],
  ],
  port: 9519,
};
```
Thanks.

**Comment by christian-bromann** [comment]
@goosewobbler any ideas?

**Comment by goosewobbler** [comment]
Hi @ellatom, so a point regarding your config that probably won't make a difference, but you should fix:

Safaridriver config is in the electron service config block, the service only accepts chromedriver opts which are passed to chromedriver internally.

As for diagnosing this, need to verify what version of the service you are running and what your logfile says after trying to run the tests on mac.

**Comment by pmerwin** [comment]
I see a similar issue with our electron app, I will try and get some steps and logs this week.

**Reply by goosewobbler** [reply]
Thanks, interesting that it's mac only. `wdio-electron-service-example` still working ok on my Mac.  I suspect some kind of issue with paths

**Comment by goosewobbler** [comment]
@pmerwin @ellatom I don't know if it's related but I did identify a [bug](https://github.com/webdriverio-community/wdio-electron-service/issues/28) to do with chromedriver versions.

Also I'm removing the session reload behaviour because it doesn't work as expected in all cases and the simplest thing to do here is to let users manage their own session state. So you won't need to specify `newSessionPerTest` any more and if you want a new session you just call `browser.reloadSession()` in a test.

**Reply by ellatom** [reply]
@goosewobbler @pmerwin I updated the command to run on Mac from this command and it worked:
"npx wdio run wdio.conf.ts path="/Applications/XXXXX.app"
to
"npx wdio run wdio.conf.ts path="/Applications/XXXXX.app/Contents/MacOS/XXXX"

**Reply by goosewobbler** [reply]
@ellatom right I see, I thought it might be path stuff.  Using appPath / appName might be able to handle this case when `electron-builder` is not found, I'll take a look at it.

**Reply by goosewobbler** [reply]
raised https://github.com/webdriverio-community/wdio-electron-service/issues/29
